src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1353,6 +1353,7 @@ def create_oracledb_connection():
                 fetch_size=fetch_size,
                 imports=udtf_configs.get("imports", None),
                 packages=udtf_configs.get("packages", None),
```

Comment:
How does ast work for this feature?

Reply:
dumping our discussion here:

dbapi is public api, which makes use of the ast and we emit ast right before returning the result dataframe back to users.

there are two ingestion modes and we emit ast differently:

1. local parquet file based approach
in this approach, we only emit ast for last operation which returns session.table() to the user, all internal operations won't emit ast. we expect the operations to take place on client env.

2. udtf
in this approach, we emit ast for udtf registration, udtf call, and resulting dataframe column selection.